rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -101,10 +111,15 @@ def __str__(self) -> str:
 
 
 class BaseAcquisitionsOrder(metaclass=ABCMeta):
-    _acquisitions: deque[AssetAcquisitionEvent]
+    # For FIFO, the time at the point of insertion is used
+    # For LIFO, the time at the point of insertion is used with a negation
+    # For HIFO, the amount of the acquisition event is used with a negation
+    # The reason for the negation is because of Python's `heapq` implementation
+    Priority = Union[FVal, Timestamp]
```

Comment:
I'll switch to a counter then

Reply:
Is the order guaranteed? Afaik the order when processing is important. Why is the timestamp not good?

Reply:
Because we can achieve the same with a counter and the order will be preserved. Plus counter better shows the purpose and doesn't contain any extra unneeded info